### PR TITLE
Prepare tests to move Buffer to Kotlin

### PR DESCRIPTION
**Pull request by swankjesse**
Our tests access package-private members of Buffer, Segment, and SegmentPool. This
changes the tests to use members of TestUtil instead. This migrates TestUtil to
Kotlin so that it can access those members.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,7 +11,6 @@ out
 lib
 reports
 
-target
 pom.xml.*
 release.properties
```
Comment (JakeWharton) [inline .gitignore]: All three can be nuked

File: okio/jvm/src/test/java/okio/BufferCursorKotlinTest.kt [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  @Parameter lateinit var bufferFactory: BufferCursorTest.BufferFactory
+
+  @Test fun acquireReadOnlyDoesNotCopySharedDataArray() {
+    val buffer = deepCopy(bufferFactory.newBuffer())
+    assumeTrue(buffer.size() > 0)
+
+    val shared = buffer.clone()
+    assertTrue(buffer.head!!.shared)
+
+    buffer.readUnsafe().use({ cursor ->
```
Comment (Egorand) [inline okio/jvm/src/test/java/okio/BufferCursorKotlinTest.kt]: supernit: `use {}`, without `()`

File: okio/jvm/src/test/java/okio/BufferCursorKotlinTest.kt [hunk 3]
```diff
@@ -0,0 +1,113 @@
+package okio
```
[the remark this answers is not in the document]
Comment (swankjesse) [reply okio/jvm/src/test/java/okio/BufferCursorKotlinTest.kt]: Fixed!

File: .gitignore (hunk 1 above)
Comment (swankjesse) [reply .gitignore]: Fixed.

File: okio/jvm/src/test/java/okio/BufferCursorKotlinTest.kt (hunk 2 above)
Comment (swankjesse) [reply okio/jvm/src/test/java/okio/BufferCursorKotlinTest.kt]: Fixed all.

**Pull request merged**